bindings/rust/extended/s2n-tls/src/cert_validation.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use s2n_tls_sys::*;
+
+use crate::{
+    callbacks::*,
+    connection::Connection,
+    error::{Error, Fallible},
+};
+use std::{pin::Pin, ptr::NonNull};
+
+pub struct CertValidationInfo {
+    info: NonNull<s2n_cert_validation_info>,
+}
+
+/// # Safety
+///
+/// Safety: s2n_cert_validation_info objects can be sent across threads
+unsafe impl Send for CertValidationInfo {}
+
+/// # Safety
+///
+/// Safety: All C methods that mutate the s2n_cert_validation_info are wrapped
+/// in Rust methods that require a mutable reference.
+unsafe impl Sync for CertValidationInfo {}
+
+impl CertValidationInfo {
+    pub(crate) fn from_ptr(info: *mut s2n_cert_validation_info) -> Result<Self, Error> {
+        let info = NonNull::new(info).ok_or(Error::INVALID_INPUT)?;
+        Ok(CertValidationInfo { info })
+    }
+
+    pub(crate) fn as_ptr(&self) -> *mut s2n_cert_validation_info {
+        self.info.as_ptr()
+    }
+
+    /// Corresponds to [s2n_cert_validation_accept].
+    pub fn accept(&self) -> Result<(), Error> {
+        unsafe { s2n_cert_validation_accept(self.as_ptr()).into_result() }?;
```

Comment:
The memory for the `s2n_cert_validation_info` object in the `CertValidationInfo` struct is owned by its `s2n_connection`. If the connection is freed before accept/reject is called, `s2n_cert_validation_info` will point to invalid memory.

This is also the case for the `s2n_cert_chain_and_key` object owned by `CertificateChainHandle`. In this case, a lifetime field was added to the struct in order to tie the struct's lifetime to its connection:
https://github.com/aws/s2n-tls/blob/414d24ccaceb20ceda9ff3ab13512485a10604da/bindings/rust/extended/s2n-tls/src/cert_chain.rs#L22

Is something like this also needed for the `CertValidationInfo` struct?